sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1197,6 +1253,8 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 		 */
 		dai_links[*link_index].nonatomic = true;
 
+		set_dailink_map(sdw_codec_ch_maps, codec_num, cpu_dai_num);
+		dai_links[*link_index].codec_ch_maps = sdw_codec_ch_maps;
```

Comment:
Is it safe to use the same ch_map for all PCMs?

Reply:
Thanks @ujfalusi use different sdw_codec_ch_maps  for each PCM now.
The change is
```patch
diff --git a/sound/soc/intel/boards/sof_sdw.c b/sound/soc/intel/boards/sof_sdw.c
index c9ece33f355d..d626fbcfc58f 100644
--- a/sound/soc/intel/boards/sof_sdw.c
+++ b/sound/soc/intel/boards/sof_sdw.c
@@ -1105,7 +1105,6 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 			      bool append_codec_type,
 			      int adr_index)
 {
-	struct snd_soc_dai_link_codec_ch_map *sdw_codec_ch_maps;
 	const struct snd_soc_acpi_link_adr *link_next;
 	struct snd_soc_dai_link_component *codecs;
 	int cpu_dai_id[SDW_MAX_CPU_DAIS];
@@ -1128,11 +1127,6 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 	if (!codecs)
 		return -ENOMEM;
 
-	sdw_codec_ch_maps = devm_kcalloc(dev, codec_num,
-					 sizeof(*sdw_codec_ch_maps), GFP_KERNEL);
-	if (!sdw_codec_ch_maps)
-		return -ENOMEM;
-
 	/* generate codec name on different links in the same group */
 	for (link_next = link; link_next && link_next->num_adr &&
 	     i < cpu_dai_num; link_next++) {
@@ -1180,6 +1174,7 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 
 	cpu_dai_index = *cpu_id;
 	for_each_pcm_streams(stream) {
+		struct snd_soc_dai_link_codec_ch_map *sdw_codec_ch_maps;
 		char *name, *cpu_name;
 		int playback, capture;
 		static const char * const sdw_stream_name[] = {
@@ -1192,6 +1187,11 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 		if (!codec_info_list[codec_index].direction[stream])
 			continue;
 
+		sdw_codec_ch_maps = devm_kcalloc(dev, codec_num,
+						 sizeof(*sdw_codec_ch_maps), GFP_KERNEL);
+		if (!sdw_codec_ch_maps)
+			return -ENOMEM;
+
 		/* create stream name according to first link id */
 		if (append_codec_type) {
 			name = devm_kasprintf(dev, GFP_KERNEL,
```